packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js

```diff
@@ -0,0 +1,44 @@
+const { packageExists } = require('../utils/package-exists');
+const webpack = packageExists('webpack') ? require('webpack') : undefined;
+const logger = require('../utils/logger');
+
+const PluginName = 'webpack-cli';
+
+class WebpackCLIPlugin {
+    constructor(options) {
+        this.options = options;
+    }
+    async apply(compiler) {
+        const compilers = compiler.compilers ? compiler.compilers : [compiler];
+
+        for (const compiler of compilers) {
+            if (this.options.progress) {
+                const { ProgressPlugin } = webpack;
```

Comment:
Let's improve this `const { ProgressPlugin } = compiler.webpack || webpack;`

First for webpack@5, second for webpack@4

We should avoid require `webpack` directly for v5, because it can create problems with monorepos and hoist